--- mregexp.c.orig	Sun Apr 11 21:18:54 2004
+++ mregexp.c	Tue Apr 13 20:11:48 2004
@@ -7,5 +7,6 @@
 #include <stdio.h>
-#include <ctype.h>
-#include <string.h>
+//#include <ctype.h>
+//#include <string.h>
 
+#include <winsock.h>
 #include <mysql.h>
@@ -25,3 +26,3 @@
 
-int strcmpi(const char *s1, const char *s2);
+//int strcmpi(const char *s1, const char *s2);
 
@@ -54,3 +55,3 @@
 		} enctab_type;
-		static enctab_type enctab[] = {
+		enctab_type enctab[] = {
 			{ "Shift_JIS"  , ONIG_ENCODING_SJIS },
@@ -132,3 +133,3 @@
 
-long long mregexp(UDF_INIT *initid,
+_int64 mregexp(UDF_INIT *initid,
 				  UDF_ARGS *args,
@@ -198,3 +199,3 @@
 	}
-	snprintf(result, 76, "mregexp-%s [%s] (oniguruma-%s)",
+	_snprintf(result, 76, "mregexp-%s [%s] (oniguruma-%s)",
 			 VERSION, def_enc, onig_version());
@@ -203,3 +204,3 @@
 }
-
+/*
 int strcmpi(const char *s1, const char *s2) {
@@ -216 +217,2 @@
 }
+*/
